jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/archive/JKubeArchiveDecompressor.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common.archive;
+
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.eclipse.jkube.kit.common.util.FileUtil;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class JKubeArchiveDecompressor {
+  private JKubeArchiveDecompressor() { }
+
+  /**
+   * Extracts a given archive file to specified target directory
+   *
+   * @param inputArchiveFile input archive file
+   * @param targetDirectory target folder where you want to extract
+   * @throws IOException in case of failure while trying to create any directory
+   */
+  public static void extractArchive(File inputArchiveFile, File targetDirectory) throws IOException {
+    if (targetDirectory.exists()) {
+      FileUtil.cleanDirectory(targetDirectory);
+    }
+    Files.createDirectory(targetDirectory.toPath());
+    if (inputArchiveFile.getName().endsWith(".tgz")) {
+      extractTarArchive(inputArchiveFile, targetDirectory.toPath());
+    } else if (inputArchiveFile.getName().endsWith(".zip")) {
+      extractZipArchive(inputArchiveFile, targetDirectory.toPath());
+    }
```

Comment:
I believe this is not the optimal way to detect the compression algorithm and choose the decompression method (also it relies on filenames being in lower-case).

CompressorStreamFactory.createCompressorInputStream(originalInput);

might be a more suitable option

Reply:
When I'm trying to pass file input stream to this method I'm getting mark not supported error:
`java.lang.IllegalArgumentException: Mark is not supported.`

Reply:
Probably requires the Archive factory. Please check documentation, but I'm sure there are ways to detect the algorithms in commons-compress without relying on the file extension.